libgadget/exchange.c

```diff
@@ -110,7 +109,7 @@ int domain_exchange(int (*layoutfunc)(int p), int do_gc) {
 
         /* determine for each rank how many particles have to be shifted to other ranks */
         plan.last = domain_find_iter_space(&plan);
-        domain_build_plan(layoutfunc, &plan);
+        domain_build_plan(layoutfunc, layout_userdata, &plan);
```

Comment:
When did plan remember layoutfunc in a plan? It feels like you can store these before the loop rather than storing them at an arbitrary point in the loop.

Is there any other place where we need the layoutfunc without needing a plan?

Reply:
I see you simply figured some of the functions does not need the layoutfunc, you are not saving the layoutfunc in the plan object. Would it be a good idea to store it there?

Reply:
That's right. I thought about it and I think I'd like to keep them separate. The layoutfunc creates the plan and doesn't need to be called after it is made.